pydoctor/test/test_epydoc2stan.py

```diff
@@ -59,6 +59,37 @@ def test_html_empty_module() -> None:
     """).strip()
     assert docstring2html(mod) == expected_html
 
+
+def test_func_undocumented_return_nothing() -> None:
+    """When the returned value is undocumented and its type is None,
+    omit the "Returns" entry from the output.
+    """
+    mod = fromText('''
+    def nop() -> None:
+        pass
+    ''')
+    func = mod.contents['nop']
+    lines = docstring2html(func).split('\n')
+    assert '<td class="fieldName">Returns</td>' not in lines
+
+
+def test_func_undocumented_return_something() -> None:
+    """When the returned value is undocumented and its type is not None,
+    include a the "Returns" entry in the output.
```

Comment:
I don't understand this docstring.... 

So if the return type is not None and is not documented, it is added as an undocumented span.

I guess that it handles this case https://pydoctor--296.org.readthedocs.build/en/296/api/pydoctor.astbuilder.ASTBuilder.html#parseFile

Can we also include `<tr class="fieldStart"><td class="fieldName">Returns</td><td colspan="2">` in the expected output?

Reply:
Yes, that is the case that it handles. I'll add a bit more detail to the docstring, in the hope that it will be easier to understand.

I added both `<td>` tags in the expected output, that indeed makes the test easier to read. However, I left out the `<tr>`, since that is not really part of what is being tested here and adding more output increases the chance that the test will require maintenance on unrelated changes.